sftp.go

```diff
@@ -227,3 +231,40 @@ func (s *StatusError) Error() string {
 func (s *StatusError) FxCode() fxerr {
 	return fxerr(s.Code)
 }
+
+// useful for tests
+func getEnabledExtensionByName(extensionName string) (sshExtensionPair, error) {
+	for _, enabledExtension := range sftpExtensions {
+		if enabledExtension.Name == extensionName {
+			return enabledExtension, nil
+		}
+	}
+	return sshExtensionPair{}, fmt.Errorf("Disabled or unsupported extension: %v", extensionName)
+}
+
+func getSupportedExtensionByName(extensionName string) (sshExtensionPair, error) {
+	for _, supportedExtension := range supportedSFTPExtensions {
+		if supportedExtension.Name == extensionName {
+			return supportedExtension, nil
+		}
+	}
+	return sshExtensionPair{}, fmt.Errorf("Unsupported extension: %v", extensionName)
+}
+
+// SetSFTPExtensions allows to customize the supported server extensions.
+// See the variable supportedSFTPExtensions for supported extensions.
+// This method accepts a slice of sshExtensionPair names for example 'hardlink@openssh.com'.
+// If an invalid extension is given an error will be returned and nothing will be changed
+func SetSFTPExtensions(extensions ...string) error {
+	var tempExtensions []sshExtensionPair
+	for _, extension := range extensions {
+		sftpExtension, err := getSupportedExtensionByName(extension)
+		if err != nil {
+			return err
+		}
+		tempExtensions = append(tempExtensions, sftpExtension)
+	}
+	sftpExtensions = make([]sshExtensionPair, len(extensions))
+	copy(sftpExtensions, tempExtensions)
```

Comment:
You do  not need to copy the slice anymore, as you are already building a new slice in `tempExtensions`.